sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -471,8 +479,15 @@ public override int GetHashCode()
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private string DebuggerDisplay => _element.DebuggerDisplay;
 
+#if !NET5_0
+        // Since this type is used in an attribute it cannot be annotated correctly through the call chain.
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = DynamicDataJsonConverterSuppression)]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL3050", Justification = DynamicDataJsonConverterSuppression)]
```

Comment:
I'm curious if this is actually correct. There's effectively a circular dependency.
* The DynamicData annotation relies on warning when the constructor is called - to guard all of the problems inside instance methods (like `WriteTo`).
* The converter attribute suppresses the warning it gets by calling the constructor - but the justification is "the parent type where this is used is annotated"

The problem is that in an app, there doesn't seem to be a direct call to the constructor or any of the static methods on `DynamicData`. I assume it's only constructed as a result of JSON deserialization - through the converter where the warning is suppressed.

I'll have to think about this some more, but currently I think:
* Reflection based JSON deserialization should produce a warning when used - so there should be some entrypoint into all of this - the first call to the deserializer. I don't know where it is... but I assume it has to be somewhere (the converters themselves won't work, without the deserialization entry point). So right now this is probably OK.
* If Azure.Core started to use source generator based JSON deserialization, then this could be a problem - because then the entry point does not produce warning, but there will be direct calls from the source generated code to the converter.

I'll look into this some more, so please don't do anything "yet", but I think the correct annotation here is to actually annotate the converter as problematic and suppress the warning on the attribute usage - given the constructor annotation of the parent type. That way any direct access to the converter would still produce a warning (the source generated code approach).

Unless there's some code inside Azure.Core which looks at the JsonConverterAttribute and uses the converter directly - then that would be a problem.